index.js

```diff
@@ -22,10 +23,16 @@ module.exports = function (opts) {
 			from: file.path,
 			to: file.path
 		}).then(function (res) {
+			var map;
 			file.contents = new Buffer(res.css);
 
 			if (res.map && file.sourceMap) {
-				applySourceMap(file, res.map.toString());
+				map = res.map.toJSON();
+				map.file = file.relative;
+				map.sources = [].map.call(map.sources, function (source) {
+					return path.join(path.dirname(file.relative), source);
+				});
```

Comment:
I linked the issue and this PR on the postcss issue (https://github.com/postcss/postcss/issues/240), the information within this PR might be useful.